docs/en/operations/settings/merge-tree-settings.md

```diff
@@ -1064,4 +1064,30 @@ Possible values:
 
 - throw, drop, rebuild
 
-Default value: throw
\ No newline at end of file
+Default value: throw
+
+## min_free_disk_bytes_to_perform_insert
+
+The minimum number of bytes that should be free in disk space in order to insert data. If the number of available free bytes is less than `min_free_disk_bytes_to_throw_insert` then an exception is thrown and the insert is not executed. Note that this setting:
+- takes into account the `keep_free_space_bytes` setting.
+- does not take into account the amount of data that will be written by the `INSERT` operation.
+- is only checked if a positive (non-zero) number of bytes is specified
+
+Possible values:
+
+- Any positive integer.
+
+Default value: 0 bytes.
+
+## min_free_disk_ratio_to_perform_insert 
+
+The minimum free to total disk space ratio to perform an `INSERT`. Must be a floating point value between 0 and 1. Note that this setting:
+- takes into account the `keep_free_space_bytes` setting.
+- does not take into account the amount of data that will be written by the `INSERT` operation.
+- is only checked if a positive (non-zero) ratio is specified
+
```

Comment:
I'd add here something like this:

Note that if both `min_free_disk_ratio_to_perform_insert` and `min_free_disk_bytes_to_perform_insert` are specified, ClickHouse will count on the value that will allow to perform inserts on a bigger amount of free memory.

Reply:
Agreed, updated!